sky/templates/ibm-ray.yml.j2

```diff
@@ -0,0 +1,148 @@
+cluster_name: {{cluster_name}}
+
+# The maximum number of workers nodes to launch in addition to the head node.
+max_workers: {{num_nodes - 1}}
+upscaling_speed: {{num_nodes - 1}}
+idle_timeout_minutes: 60
+
+
+provider:
+  type: external
+  module: sky.skylet.providers.ibm.IBMVPCNodeProvider
+  region: {{region}}
+  availability_zone: {{zones}}
+  # Keep (otherwise cannot reuse when re-provisioning).
+  # teardown(terminate=True) will override this.
+  cache_stopped_nodes: True
+  iam_api_key: {{iam_api_key}}
+  resource_group_id: {{resource_group_id}}
+  # Disable launch config check for worker nodes as it can cause resource
+  # leakage.
+  # Reference: https://github.com/ray-project/ray/blob/840215bc09e942b50cad0ab2db96a8fdc79217c1/python/ray/autoscaler/_private/autoscaler.py#L1101
+  # The upper-level SkyPilot code has make sure there will not be resource
+  # leakage.
+  disable_launch_config_check: true
+
+auth:
+  ssh_user: ubuntu
+  ssh_private_key: {{ssh_private_key}}
+
+
+available_node_types:
+  ray_head_default:
+    resources: {{instance_resources}}
+    node_config:
+      image_id: {{image_id}}  
+      boot_volume_capacity: {{disk_capacity}}
+      volume_tier_name: general-purpose
+      instance_profile_name: {{instance_type}}
+
+      {% if use_spot %}
+      InstanceMarketOptions:
+          MarketType: spot
+
+      {% endif %}
+{% if num_nodes > 1 %}
+  ray_worker_default:
+    min_workers: {{num_nodes - 1}}
+    max_workers: {{num_nodes - 1}}
+    resources: {{worker_instance_resources}}
+    node_config:
+      image_id: {{image_id}} 
```

Comment:
Can we write down the example `image_id` a user can specify in the doc page like the following comments?
https://github.com/skypilot-org/skypilot/blob/48bcfe89a3c11b879a111d4d11a4dc0f8703dd0a/docs/source/reference/yaml-spec.rst#L101-L117

Reply:
As far as i know: 
- No user can create a VPC image and share it publicly, i.e. publish it on the public VPC image catalog, so I can't create and share images as suggested. I'll look further into it and will update if something's changed. 
- IBM doesn't have a public VPC image containing CUDA / ML libraries (e.g. pytorch).
- `--image-id` flag is supported, but to take advantage of it a user should create a private VPC image beforehand. 

To create a custom VPC image the user can use an official plugin: [packer](https://www.ibm.com/cloud/blog/use-ibm-packer-plugin-to-create-custom-images-on-ibm-cloud-vpc-infrastructure), or a quicker, smaller tool: [vpc-img-inst](https://github.com/IBM/vpc-img-inst). It can be as simple as `vpc-img-inst -f cuda -r us-east` to create a cuda image based on ubuntu-22-04 in the us-east region.

Should i add some version of the above as documentation?

Reply:
Ahh, sorry for the confusion. We don’t need to support tagging the public images. I am trying to say that we should mention about the IBM cloud here, so the user knows they can provide their own image here. Also, we may want to provide a example string for what format should that self provided imageid looks like, e.g. the gcp image id should be `projects/project-id/xxxx/<image-name>`; while the AWS is `ami-xxxxxxxx`